main.tf

```diff
@@ -31,62 +41,82 @@ resource "aws_dynamodb_table" "main" {
       hash_key           = global_secondary_index.value.hash_key
       range_key          = lookup(global_secondary_index.value, "range_key", null)
       projection_type    = global_secondary_index.value.projection_type
       non_key_attributes = lookup(global_secondary_index.value, "non_key_attributes", null)
     }
   }
-  dynamic "point_in_time_recovery" {
-    for_each = length(keys(var.point_in_time_recovery)) == 0 ? [] : [var.point_in_time_recovery]
-    content {
-      enabled = point_in_time_recovery.value.enabled
-    }
+
+  point_in_time_recovery {
+    enabled = var.point_in_time_recovery_enabled
   }
+
   dynamic "replica" {
     for_each = length(keys(var.replica)) == 0 ? [] : [var.replica]
     content {
       region_name = replica.value.region_name
       kms_key_arn = lookup(replica.value, "kms_key_arn", null)
     }
   }
+
   restore_source_name    = var.restore_source_name
   restore_to_latest_time = var.restore_to_latest_time
   restore_date_time      = var.restore_date_time
   stream_enabled         = var.stream_enabled
   stream_view_type       = var.stream_view_type
-  dynamic "server_side_encryption" {
-    for_each = length(keys(var.server_side_encryption)) == 0 ? [] : [var.server_side_encryption]
-    content {
-      enabled     = server_side_encryption.value.enabled
-      kms_key_arn = lookup(server_side_encryption.value, "kms_key_arn", "alias/aws/dynamodb")
-    }
+
+  server_side_encryption {
+    enabled     = var.sse_enabled
+    kms_key_arn = var.sse_enabled && var.sse_kms_key_arn == null && var.create_sse_kms_key == false ? local.default_ddb_kms_key : (var.sse_enabled && var.sse_kms_key_arn == null && var.create_sse_kms_key ? try(aws_kms_key.ddbsse[0].arn, null) : (var.sse_enabled && var.sse_kms_key_arn != null && var.create_sse_kms_key == false ? var.sse_kms_key_arn : null))
   }
+
+  ## Interpretation of above:
+  ## Use default kms key if; sse is enabled, sse-kms-key-arn not provided and key creation not specified, i.e condition is false
+  ## Create key if; sse is enabled, sse-kms-key-arn not provided and key creation is specified, i.e condition is true
+  ## Use provided key if; sse is enabled, sse-kms-key-arn value is provided and key creation is not specified, i.e condition is false
+  ## Otherwise the value is `null`
+
   timeouts {
     create = lookup(var.timeouts, "create", "10m")
     update = lookup(var.timeouts, "update", "60m")
     delete = lookup(var.timeouts, "delete", "10m")
   }
+
   table_class = var.table_class
   tags        = var.tags
 
   lifecycle {
     ignore_changes = [
       read_capacity,
-      write_capacity
+      write_capacity,
     ]
   }
+}
+
+resource "aws_kms_key" "ddbsse" {
+  count                   = var.create_sse_kms_key ? 1 : 0
+  description             = "KMS Key for ${local.name} DynamoDB SSE"
+  policy                  = data.aws_iam_policy_document.main.json
```

Comment:
We ideally want to also allow for a custom KMS policy here, can you add this to the list of the unreleased?